From c37a0a0fb3ae26d59cfd6352e56b5ff27e796ced Mon Sep 17 00:00:00 2001
In-Reply-To: <3cabed9e-3949-93cc-2c9c-500a9cd9d4cd@siemens.com>
References: <3cabed9e-3949-93cc-2c9c-500a9cd9d4cd@siemens.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Tue, 14 May 2019 00:08:42 +0200
Subject: [PATCH] sha1dc: update from upstream
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Update sha1dc from the latest version by the upstream
maintainer[1]. See 07a20f569b ("Makefile: fix unaligned loads in
sha1dc with UBSan", 2019-03-12) for the last update.

This fixes an issue where HP-UX IA64 was wrongly detected as a
Little-endian instead of a Big-endian system, see [2] and [3].

1. https://github.com/cr-marcstevens/sha1collisiondetection/commit/855827c583bc30645ba427885caa40c5b81764d2
2. https://public-inbox.org/git/603989bd-f86d-c61d-c6f5-fb6748a65ba9@siemens.com/
3. https://github.com/cr-marcstevens/sha1collisiondetection/pull/50

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---

n Thu, May 09 2019, Osipov, Michael wrote:

> Hey there,
>
> Am 2019-05-09 um 09:32 schrieb Ævar Arnfjörð Bjarmason:
>>
>> On Wed, May 08 2019, Osipov, Michael wrote:
>>
>>> Hi folks,
>>
>> Hi see Documentation/SubmittingPatches for how to submit patches inline
>> instead of as attachments.
>
> Do you want me to resend the configure.ac change as per wiki article?
> I can also create a PR on GitHub. I am happy with both as long as I
> don't have to retain the patch for myself only ;-)

Yeah that patch to git.git should be done separately. I see your PR
went in upstream, here's a patch to update our code to match.

> That'd be much more work to extend configure.ac for that because is a
> runtime check. Since there are no real products vailable on x86 for
> HP-UX I'd neglect that. Our HPE salesman told us that this will be
> available somewhere in the future. So, I think this is very good for
> now.

Sure, makes sense. I'm not familiar with HP/UX. So just thought I'd
ask.

 sha1collisiondetection | 2 +-
 sha1dc/sha1.c          | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sha1collisiondetection b/sha1collisiondetection
index 16033998da..855827c583 160000
--- a/sha1collisiondetection
+++ b/sha1collisiondetection
@@ -1 +1 @@
-Subproject commit 16033998da4b273aebd92c84b1e1b12e4aaf7009
+Subproject commit 855827c583bc30645ba427885caa40c5b81764d2
diff --git a/sha1dc/sha1.c b/sha1dc/sha1.c
index 5931cf25d5..9d3cf81d4d 100644
--- a/sha1dc/sha1.c
+++ b/sha1dc/sha1.c
@@ -93,7 +93,7 @@
 #define SHA1DC_BIGENDIAN
 
 /* Not under GCC-alike or glibc or *BSD or newlib or <processor whitelist> */
-#elif (defined(_AIX))
+#elif (defined(_AIX) || defined(__hpux))
 
 /*
  * Defines Big Endian on a whitelist of OSs that are known to be Big
-- 
2.21.0.1020.gf2820cf01a

